src/Avalonia.Documents/Documents/InlineCollection.cs

```diff
@@ -0,0 +1,116 @@
+﻿using System;
+using System.Collections;
+using System.Text;
+using Avalonia.Collections;
+using Avalonia.Metadata;
+
+namespace Avalonia.Documents
+{
+    /// <summary>
+    /// A collection of <see cref="Inline"/>s.
+    /// </summary>
+    [WhitespaceSignificantCollection]
+    public class InlineCollection : AvaloniaList<Inline>, IList
+    {
+        bool _isNull;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="InlineCollection"/> class.
+        /// </summary>
+        public InlineCollection()
+        {
+            ResetBehavior = ResetBehavior.Remove;
+            this.ForEachItem(
+                x =>
+                {
+                    x.Invalidated += Invalidate;
+                    Invalidate();
+                },
+                x =>
+                {
+                    x.Invalidated -= Invalidate;
+                    Invalidate();
+                },
+                () => throw new NotSupportedException());
+        }
+
+        /// <summary>
+        /// Gets a string representation of the inlines.
+        /// </summary>
+        public string Text
+        {
+            get
+            {
+                if (Count == 0)
+                {
+                    return _isNull ? null : string.Empty;
+                }
+                else if (Count == 1)
+                {
+                    return ((IHasText)this[0]).Text ?? string.Empty;
```

Comment:
If TextBlock contains only <Line Break /> - this code throw exception. May be use "is IHasText"?

Reply:
Nice find, as always! :)